### PR TITLE
Initial role PR

### PR DESCRIPTION
**Pull request by haslersn**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+There also is a role variable `systemd_network_copy_files` which
+simply takes a list of which each element is passed to the Ansible
+[copy module](https://docs.ansible.com/ansible/latest/modules/copy_module.html).
+
+### WireGuard client example
+
+`systemd_network_copy_files` can for example be used for WireGuard key
+files, as illustrated in the following example.
+Installing WireGuard is out of the scope of this role and can be done using
+[our wireguard role]([https:](https://github.com/stuvusIT/wireguard)).
```
Comment (SlothOfAnarchy) [inline README.md]:
```suggestion
[our wireguard role](https://github.com/stuvusIT/wireguard).
```

File: README.md [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+
+## Upload extra files
+
+There also is a role variable `systemd_network_copy_files` which
+simply takes a list of which each element is passed to the Ansible
+[copy module](https://docs.ansible.com/ansible/latest/modules/copy_module.html).
+
+### WireGuard client example
+
+`systemd_network_copy_files` can for example be used for WireGuard key
```
Comment (SlothOfAnarchy) [inline README.md:240]: Why not implement this functionality in the wireguard role?

File: meta/main.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+---
+galaxy_info:
+  author: haslersn
+  description: Network configuration using systemd.network; see https://www.freedesktop.org/software/systemd/man/systemd.network.html
+  company: stuvus (https://stuvus.uni-stuttgart.de)
+
+  license: AGPLv3
+
+  min_ansible_version: 2.4
```
Comment (SlothOfAnarchy) [inline meta/main.yml]:
```suggestion
  min_ansible_version: 2.6
```
See https://docs.ansible.com/ansible/latest/user_guide/playbooks_filters.html#dict-filter

File: tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    loop_var: netdev
+  notify:
+    - Reload systemd configuration
+    - Restart systemd-networkd
+
+- name: Instantiate systemd-networkd networks
+  template:
+    src: network.j2
+    dest: /etc/systemd/network/{{ network.key }}.network
+    owner: systemd-network
```
Comment (SlothOfAnarchy) [inline tasks/main.yml:19]: mode?

**Review by SlothOfAnarchy — CHANGES_REQUESTED** [review]
Please add a LICENCE file so that GitHub will correctly index it.
Otherwise a very clean and well-explained role.

**Comment by haslersn** [comment]
> Please add a LICENCE file so that GitHub will correctly index it.
> Otherwise a very clean and well-explained role.

The repo was already created with LICENSE file.

File: tasks/main.yml (hunk 4 above)
Comment (haslersn) [reply tasks/main.yml:19]: The default mode on Debian 10 (0644) is fine. Should I explicitly state it?

File: README.md (hunk 2 above)
Comment (haslersn) [reply README.md:240]: The wireguard role just installs WireGuard. WireGuard is configured using the systemd-network role.

File: tasks/main.yml (hunk 4 above)
Comment (SlothOfAnarchy) [reply tasks/main.yml:19]: I thought maybe you forgot it, because you explicitly set it for the netdev files

**Review by SlothOfAnarchy — APPROVED** [review]

**Pull request merged**